### ISSUE TITLE
Cuncurrent subscribe request are lost

### ISSUE THREAD

**Issue by EnricoBeltramo** [nanomq/nanomq#1802]
**Describe the bug**
I use a script in JS based on mqttJS client already validated with other brokers. I noticed that if I send multiple parallel request of subscription from same client (using mqtt over websocket, I didn't tested mqtt raw), without waiting before the response, some of subscription request are lost. With other brokers that is managed correctly.

**Expected behavior**
Broker able to manage multiple subscriptions requests in parallel.

**Actual Behavior**
Describe what occurred.
If I send multiple request in parallel, some are lost

**To Reproduce**
If possible include actual reproduction test code here.
Minimal C test cases are perferred.
- In js, with mqttjs, open a connection
- From client send multiple request in parallel without waiting response before, i.e.:
client.subscribe(subscribetopic1, options)
client.subscribe(subscribetopic2, options)
client.subscribe(subscribetopic3, options)
client.subscribe(subscribetopic3, options)

**Environment Details**
 - NanoMQ version: docker: emqx/nanomq:0.21-full
 - Operating system and version: win
 - Compiler and language used: nodejs 20
 - testing scenario: js client on host, broker in Docker

**Client SDK**
If possible include the mqtt sdk you used to connect to nanomq
Minimal C test cases are perferred.
sdk: mqttjs
code: see previous response

**Additional context**

**Comment by JaylinYu** [comment]
WebSocket is inherited from NNG, definitely not on the same performance level as TCP/TCL
any related logs?

**Comment by EnricoBeltramo** [comment]
Ok, I'm working to change my code and do subscription sequentially. I'm just surprised that the original code was woking for one year with different brokers and I didn't had this issue, is it that expected?
I haven't many log because when the subscription is lost I haven't any repsonse, like the request never arrive to broker or broker ignored it.

**Comment by JaylinYu** [comment]
No, definitely not expected. What I mean log, is the nanomq.log, not your client log.
If even nanomq doesn't prints any log, then websocket transport of NNG is the one being suspicious, would need your mqtt.js code to continue debugging

**Comment by alexanderko** [comment]
Hello @JaylinYu . I have got similar issue an have more details here. 

It looks like sequential commands sent to MQTT.js client on a Browser over WebSocket connection go into single WebSocket packet. NanoMQ handles only first command from this WS packet and ignores the rest. Confirmed with version `nanomq-0.22.1-windows-x86_64` running on Windows Server 2016. 

Can be reproduced from the browser having MQTTjs connection available with the following code
```javascript
([1, 2, 3]).forEach(id => client.publish("test-topic", ""+id))
```

Or no-loop equivalent 
```javascript
client.publish('test-topic', '1');
client.publish('test-topic', '2');
client.publish('test-topic', '3');
```

This produces the next WebSocket packet. 
![mqtt-ws-envelope](https://github.com/user-attachments/assets/11d0974f-c7c6-4767-923e-66c1fad2a646)

The client subscribed to topic `test-topic` will receive only first message from the series.

This issue can be reproduced only in a Browser. Broker will handle all messages having the same snippet running in Node.js. I suppose because Node.js uses external ws package to work with WebSocket and this implementation flushes every message into individual WS packet.

**Comment by JaylinYu** [comment]
Probably the same bug as https://github.com/nanomq/nanomq/issues/1844
Fixed, reopen if you got any new issues @alexanderko
